source/Octopus.Cli/Commands/ApiCommand.cs

```diff
@@ -127,6 +129,21 @@ public async Task Execute(string[] commandLineArguments)
             var endpoint = string.IsNullOrWhiteSpace(ApiKey)
                 ? new OctopusServerEndpoint(ServerBaseUrl)
                 : new OctopusServerEndpoint(ServerBaseUrl, ApiKey);
+            
+            /*
+             * There may be a delay between the completion of a large file upload and when Octopus responds
+             * to finish the HTTP connection. This delay can be several minutes. During this time, no traffic is
+             * sent, and some networking infrastructure will close the connection. For example, Azure VMs will
+             * close idle connections after 4 minutes, and AWS VMs will close them after 350 seconds. The
+             * TCP keepalive option will ensure that the connection is not idle at the end of the file upload.
+             *
+             * This is the bug that explains why this doesn't work with .NET Core:
+             * https://github.com/dotnet/corefx/issues/26013
```

Comment:
@mcasperson given this has no effect in netcore should we add a compile directive to this and not even exposed this argument?

Reply:
Thats a good idea, I've added a check to only expose it for the full .NET builds